src/sentry/lang/java/plugin.py

```diff
@@ -92,6 +92,7 @@ def process_exception(self, exception):
     def process_frame(self, processable_frame, processing_task):
         frame = processable_frame.frame
         raw_frame = dict(frame)
+        release = super().get_release()
```

Comment:
we get this from the db, but theoretically, we could also parse the release field from the event data, but I'm not sure what's faster

Reply:
Does reading from DB introduce a lag, where updating the release package in the SDK only takes effect later on? Might be confusing to debug this way.

Reply:
Where on the event is the release package? Or would we just take the release from the event data to look up the release object containing the package?

Reply:
> Where on the event is the release package? Or would we just take the release from the event data to look up the release object containing the package?

the latter, yeah https://github.com/getsentry/sentry/blob/6f3a2e4a54a2efc26534c2d9ab86a3e74eec6fec/src/sentry/stacktraces/processing.py#L136-L153

> Does reading from DB introduce a lag, where updating the release package in the SDK only takes effect later on?

not sure I got what you mean?

Reply:
> Does reading from DB introduce a lag, where updating the release package in the SDK only takes effect later on?

You can ignore this question since it's only referencing the release and not sending the package in the event.

I presume the release is being created by sentry-cli while uploading things like source context or proguard mapping files. Do we need to specify the package as a param to sentry-cli in SAGP and maven plugin?

Reply:
Ah looks like this is created on the fly when sending events, see https://docs.sentry.io/platforms/android/configuration/releases/

So package would be in the release like this: `my.project.name@2.3.12+1234`

So if I understand correctly, we'd need something like https://github.com/getsentry/sentry-java/blob/54d51f22dc565154f868848d1118380444c45491/sentry-android-core/src/main/java/io/sentry/android/core/AndroidOptionsInitializer.java#L300-L308 for Spring Boot as well, correct?

Reply:
I guess since Spring Boot applications are usually not obfuscated, the easier approach would be to simply set `in-app-includes` by default.

Reply:
Ah TIL: https://github.com/getsentry/sentry-java/blob/7cdf1216cba07a8bc4adb24ca9356bb8231ee724/sentry-spring-boot/src/main/java/io/sentry/spring/boot/InAppIncludesResolver.java#L22

Sorry for the spam here.

Reply:
the release is created on the fly when a first event for that release gets processed. There's a possibility to announce releases through sentry-cli, but we don't offer that in the gradle/maven plugin yet.

So, this logic is not gonna work for the first event in the release, because there's no release in the db yet, but I think it's fine? I didn't want to specify `create=True` because not sure of the consequences doing it here. Or we could switch to just parsing the `event.release` field

Reply:
so to sum it up - we don't need to do anything on the SDK side, it's just a nice addition for obfuscated apps. There was always a possibility to specify a custom stacktrace rule for marking your own frames as in_app, and we're just automatically doing that here.